spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/pulsar/PulsarPropertiesMapper.java

```diff
@@ -195,6 +197,13 @@ private void customizePulsarContainerListenerProperties(PulsarContainerPropertie
 		map.from(properties::isObservationEnabled).to(containerProperties::setObservationEnabled);
 	}
 
+	<T> void customizeConcurrentPulsarListenerContainerFactory(
```

Comment:
I just noticed that the listener containers on the framework side are inconsistent w/ the `concurrency` attribute between reactive and imperative. I created [this issue to simplify the latter](https://github.com/spring-projects/spring-pulsar/issues/820). 
If you wanted to handle that issue first it would then simplify this PR so that we would not need this new `customizeConcurrentPulsarListenerContainerFactory` method and could simply map the property in `customizePulsarContainerListenerProperties` instead.

Reply:
Sure, I can pick up that issue on Spring Pulsar side. So basically this PR is then blocker until https://github.com/spring-projects/spring-pulsar/issues/820 is resolved.

Reply:
> Sure, I can pick up that issue on Spring Pulsar side. So basically this PR is then blocker until [spring-projects/spring-pulsar#820](https://github.com/spring-projects/spring-pulsar/issues/820) is resolved.

Thanks @vpavic . Yeh, I think we should do the spring-pulsar one first.